org.scala-ide.sdt.core/src/org/scalaide/ui/editor/DecoratedInteractiveEditor.scala

```diff
@@ -23,28 +23,30 @@ trait DecoratedInteractiveEditor extends ISourceViewerEditor {
    * Update annotations on the editor from a list of IProblems
    */
   def updateErrorAnnotations(errors: List[IProblem], cu: ICompilationUnit) {
-    val newAnnotations: Map[Annotation, Position] = (for (e <- errors) yield {
-      val annotation = new ProblemAnnotation(e, cu) // no compilation unit
-      val position = new Position(e.getSourceStart, e.getSourceEnd - e.getSourceStart + 1)
-      (annotation, position)
-    })(breakOut)
-
-    annotationModel.replaceAnnotations(previousAnnotations, newAnnotations)
-    previousAnnotations = newAnnotations.keys.toList
-
-    // This shouldn't be necessary in @dragos' opinion. But see #84 and
-    // http://stackoverflow.com/questions/12507620/race-conditions-in-annotationmodel-error-annotations-lost-in-reconciler
-    val presViewer = getViewer
-    if (presViewer.isInstanceOf[ITextViewerExtension2]) {
-        // TODO: This should be replaced by a better modularization of semantic highlighting PositionsChange
-        val newPositions = newAnnotations.values
-        def end (x:Position) = x.offset + x.length - 1
-        val taintedBounds : (Int, Int) = ((Int.MaxValue, 0) /: newPositions) {(acc, p1) => (Math.min(acc._1, p1.offset), Math.max(acc._2, end(p1)))}
-        val taintedLength = (taintedBounds._2 - taintedBounds._1 +1)
-
-        DisplayThread.asyncExec { presViewer.asInstanceOf[ITextViewerExtension2].invalidateTextPresentation(taintedBounds._1, taintedLength) }
-    } else {
-        DisplayThread.asyncExec { getViewer.invalidateTextPresentation() }
+    annotationModel foreach { model ⇒
```

Comment:
If you use this version:
`def updateErrorAnnotations(errors: List[IProblem], cu: ICompilationUnit): Unit = annotationModel foreach { model ⇒` all those lines will be left unchanged (which helps with reviews and file history browsing). But that's just a side-note.